### PR TITLE
Compile on ROS2 Foxy

### PR DESCRIPTION
**Pull request by henningkayser**
Once #226 is merged, I will branch off `foxy` and retarget this PR, generate docker images and trigger CI.

### CODE REVIEW

File: moveit_core/constraint_samplers/test/pr2_arm_kinematics_plugin.cpp [hunk 1]
```diff
@@ -97,6 +97,12 @@ PR2ArmIKSolver::PR2ArmIKSolver(const urdf::ModelInterface& robot_model, const st
   active_ = pr2_arm_ik_.init(robot_model, root_frame_name, tip_frame_name);
 }
 
+void PR2ArmIKSolver::updateInternalDataStructures()
+{
+  // TODO: move (re)allocation of any internal data structures here
```
Comment (tylerjw) [inline moveit_core/constraint_samplers/test/pr2_arm_kinematics_plugin.cpp:102]: TODO should have a username or issue number

Comment (tylerjw) [inline moveit_core/constraint_samplers/test/pr2_arm_kinematics_plugin.cpp:102]: TODO should have a username or issue number

File: moveit_core/robot_state/src/robot_state.cpp [hunk 2]
```diff
@@ -165,7 +165,7 @@ void RobotState::copyFrom(const RobotState& other)
             (robot_model_->getVariableCount() * (1 + ((has_velocity_ || has_acceleration_ || has_effort_) ? 1 : 0) +
                                                  ((has_acceleration_ || has_effort_) ? 1 : 0)) +
              nr_doubles_for_dirty_joint_transforms);
-    memcpy(variable_joint_transforms_, other.variable_joint_transforms_, bytes);
+    memcpy((char*)variable_joint_transforms_, (char*)other.variable_joint_transforms_, bytes);
```
Comment (tylerjw) [inline moveit_core/robot_state/src/robot_state.cpp:168]: Use a cpp style cast.

File: moveit_core/constraint_samplers/test/pr2_arm_kinematics_plugin.cpp (hunk 1 above)
Comment (henningkayser) [reply moveit_core/constraint_samplers/test/pr2_arm_kinematics_plugin.cpp:102]: This function has been removed by mistake and I just reverted it, the original TODO also exists like that in MoveIt 1. Not sure if it makes sense to change this here

File: moveit_core/robot_state/src/robot_state.cpp (hunk 2 above)
Comment (henningkayser) [reply moveit_core/robot_state/src/robot_state.cpp:168]: See https://github.com/ros-planning/moveit/pull/2180#discussion_r448487601. I'd like to keep the diff minimal or we should change it for all versions. I should update it to void, though.

**Comment by henningkayser** [comment]
Travis succeeded, Github just doesn't get updated (https://travis-ci.com/github/ros-planning/moveit2/builds/179287457)

**Comment by henningkayser** [comment]
Cleaned up redundant commits

**Pull request merged**